include/s3select.h

```diff
@@ -2240,7 +2253,17 @@ class base_s3object
 
     for (auto& p : m_projections)
     {
+      if(p->is_statement_contain_star_operation())
+      {
+        set_star_true();
+      }
       p->traverse_and_apply(m_sa, m_s3_select->get_aliases(), m_s3_select->is_json_query());
+
+      if(m_s3_select->is_json_query() && p->is_column())  {
+        json_keys.push_back(p->get_json_key());
+      } else if(m_s3_select->is_json_query() && !p->is_column())  {
+        json_keys.push_back("nil");
```

Comment:
in case it is not a column reference 
it should not present a `nil` as the object name 
it should present a column-number ` _{projection-location}`

Reply:
Yes, nil can be omitted. There is no need for the else part. Using a lable variable for that purpose.